src/Microsoft.TemplateEngine.TemplateLocalizer.Core/KeyExtractors/PropertyNameBasedKeyCreator.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Core.KeyExtractors
+{
+    /// <summary>
+    /// Creates a key for a given element using the name of the element.
+    /// </summary>
+    internal sealed class PropertyNameBasedKeyCreator : IJsonKeyCreator
+    {
+        /// <inheritdoc/>
+        public string CreateKey(JsonElement element, string? elementName, string? parentElementName, int indexInParent)
```

Comment:
do we need 2-4 parameter? can we get it from the element?

Reply:
`JsonElement` doesn't provide API to move up in json hierarchy.